### ISSUE TITLE
Appearance switch button and custom css work with dev, but fail with preview

### ISSUE THREAD

**Issue by GVenusLeo** [vuejs/vitepress#2075]
### Describe the bug

When I use `yarn docs:dev` to view my site, it works. But when I used `yarn docs:build` and `yarn docs:preview`, the site's appearance switch button didn't work when clicked, and the custom css didn't work either.

### Expected behavior

Appearance switch button and custom css work.

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 AMD Ryzen 7 4700U with Radeon Graphics
    Memory: 8.17 GB / 15.37 GB
  Binaries:
    Node: 18.15.0 - C:\CodeSDK\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.5.0 - C:\CodeSDK\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1265.0), Chromium (110.0.1587.69)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    vitepress: ^1.0.0-alpha.52 => 1.0.0-alpha.52
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.dev).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

**Comment by amir20** [comment]
I saw this happen when upgrading to version 22. There is a JavaScript error in production which is breaking all js. Reverting to previous version fixes it. 

I'll post the error when I get on my laptop.

**Comment by amir20** [comment]
Actually #2073 and #2074 are duplicates and have the error.

**Comment by yyx990803** [comment]
should be the same and fixed in 905f58b
